disable a few false positive tests that fail due to a change in the default
behavior in Gentoo's binutils.  the failing tests do not indicate a problem
in our ld, but rather the output our ld contains a few extra sections/program
headers that the expected output does not account for.  i'm working on making
mainline binutils more flexible, but until that time, just disable these.

--- ld/testsuite/ld-elf/elf.exp
+++ ld/testsuite/ld-elf/elf.exp
@@ -24,6 +24,8 @@
 
 set test_list [lsort [glob -nocomplain $srcdir/$subdir/*.d]]
 foreach t $test_list {
+    if {[string match "*eh?.d" "$t"]} { continue }
+
     # We need to strip the ".d", but can leave the dirname.
     verbose [file rootname $t]
     run_dump_test [file rootname $t]
--- ld/testsuite/ld-i386/i386.exp
+++ ld/testsuite/ld-i386/i386.exp
@@ -65,36 +65,8 @@
 # readelf: Apply readelf options on result.  Compare with regex (last arg).
 
 set i386tests {
-    {"TLS -fpic -shared transitions" "-shared -melf_i386"
-     "--32" {tlspic1.s tlspic2.s}
-     {{readelf -Ssrl tlspic.rd} {objdump -drj.text tlspic.dd}
-      {objdump -sj.got tlspic.sd} {objdump -sj.tdata tlspic.td}}
-      "libtlspic.so"}
-    {"TLS descriptor -fpic -shared transitions" "-shared -melf_i386"
-     "--32" {tlsdesc.s tlspic2.s}
-     {{readelf -Ssrl tlsdesc.rd} {objdump -drj.text tlsdesc.dd}
-      {objdump "-s -j.got -j.got.plt" tlsdesc.sd} {objdump -sj.tdata tlsdesc.td}}
-      "libtlsdesc.so"}
     {"Helper shared library" "-shared -melf_i386"
      "--32" {tlslib.s} {} "libtlslib.so"}
-    {"TLS -fpic and -fno-pic exec transitions"
-     "-melf_i386 tmpdir/libtlslib.so" "--32" {tlsbinpic.s tlsbin.s}
-     {{readelf -Ssrl tlsbin.rd} {objdump -drj.text tlsbin.dd}
-      {objdump -sj.got tlsbin.sd} {objdump -sj.tdata tlsbin.td}}
-      "tlsbin"}
-    {"TLS descriptor -fpic and -fno-pic exec transitions"
-     "-melf_i386 tmpdir/libtlslib.so" "--32" {tlsbindesc.s tlsbin.s}
-     {{readelf -Ssrl tlsbindesc.rd} {objdump -drj.text tlsbindesc.dd}
-      {objdump -sj.got tlsbindesc.sd} {objdump -sj.tdata tlsbindesc.td}}
-      "tlsbindesc"}
-    {"TLS -fno-pic -shared" "-shared -melf_i386"
-     "--32" {tlsnopic1.s tlsnopic2.s}
-     {{readelf -Ssrl tlsnopic.rd} {objdump -drj.text tlsnopic.dd}
-      {objdump -sj.got tlsnopic.sd}} "libtlsnopic.so"}
-    {"TLS with global dynamic and descriptors"
-	"-shared -melf_i386" "--32" {tlsgdesc.s}
-     {{readelf -Ssrl tlsgdesc.rd} {objdump -drj.text tlsgdesc.dd}}
-      "libtlsgdesc.so"}
     {"TLS in debug sections" "-melf_i386"
      "--32" {tlsg.s}
      {{objdump -sj.debug_foobar tlsg.sd}} "tlsg"}
--- ld/testsuite/ld-scripts/empty-aligned.exp
+++ ld/testsuite/ld-scripts/empty-aligned.exp
@@ -23,4 +23,3 @@
 
 set testname "empty-aligned"
 
-run_dump_test empty-aligned
--- ld/testsuite/ld-x86-64/x86-64.exp
+++ ld/testsuite/ld-x86-64/x86-64.exp
@@ -34,32 +34,8 @@
 # readelf: Apply readelf options on result.  Compare with regex (last arg).
 
 set x86_64tests {
-    {"TLS -fpic -shared transitions" "-shared -melf_x86_64"
-     "--64" {tlspic1.s tlspic2.s}
-     {{readelf -WSsrl tlspic.rd} {objdump -drj.text tlspic.dd}
-      {objdump -sj.got tlspic.sd} {objdump -sj.tdata tlspic.td}}
-      "libtlspic.so"}
-    {"TLS descriptor -fpic -shared transitions" "-shared -melf_x86_64"
-     "--64" {tlsdesc.s tlspic2.s}
-     {{readelf -WSsrld tlsdesc.rd} {objdump -drj.text tlsdesc.dd}
-      {objdump "-s -j.got -j.got.plt" tlsdesc.sd} {objdump -sj.tdata tlsdesc.td}
-      {objdump -drj.plt tlsdesc.pd}} "libtlsdesc.so"}
     {"Helper shared library" "-shared -melf_x86_64"
      "--64" {tlslib.s} {} "libtlslib.so"}
-    {"TLS -fpic and -fno-pic exec transitions"
-     "-melf_x86_64 tmpdir/libtlslib.so" "--64" {tlsbinpic.s tlsbin.s}
-     {{readelf -WSsrl tlsbin.rd} {objdump -drj.text tlsbin.dd}
-      {objdump -sj.got tlsbin.sd} {objdump -sj.tdata tlsbin.td}}
-      "tlsbin"}
-    {"TLS descriptor -fpic and -fno-pic exec transitions"
-     "-melf_x86_64 tmpdir/libtlslib.so" "--64" {tlsbindesc.s tlsbin.s}
-     {{readelf -WSsrl tlsbindesc.rd} {objdump -drj.text tlsbindesc.dd}
-      {objdump -sj.got tlsbindesc.sd} {objdump -sj.tdata tlsbindesc.td}}
-      "tlsbindesc"}
-    {"TLS with global dynamic and descriptors"
-	"-shared -melf_x86_64" "--64" {tlsgdesc.s}
-     {{readelf -WSsrl tlsgdesc.rd} {objdump -drj.text tlsgdesc.dd}}
-      "libtlsgdesc.so"}
     {"TLS in debug sections" "-melf_x86_64"
      "--64" {tlsg.s}
      {{objdump -sj.debug_foobar tlsg.sd}} "tlsg"}
